tests/page-objects/reports/reports.po.js

```diff
@@ -3,8 +3,34 @@ const helper = require('../../helper');
 const itemSummary = '#reports-content .report-body .item-summary';
 const reportBody = '#reports-content .report-body';
 const reportBodyDetails = '#reports-content .report-body .details';
+const datePickerStart = element(by.css('.daterangepicker [name="daterangepicker_start"]'));
+const datePickerEnd = element(by.css('.daterangepicker [name="daterangepicker_end"]'));
+const dateFilter = element(by.css('#date-filter'));
+const reportListID = '#reports-list';
+const list = element(by.css(reportListID));
+const listLoader =  element(by.css(`${reportListID} .loader`));
+const firstReport = element(by.css(`${reportListID} li:first-child`));
+const allReports = element.all(by.css(`${reportListID} li`));
```

Comment:
All the existing code (eg: line 5) just has a constant for the CSS selector, because IIRC executing `element.all` too early caused issues. Do you know if this has changed? The test is passing so I assume it's working as expected...

Reply:
Would doing this work or would it be better to declare a function that returns the result of `.all`? @garethbowen  @dianabarsan 

``` 
module.exports = {
  allReports: element.all(by.css(`${reportListID} li`))
}```

Reply:
`allReports: element.all(by.css(`${reportListID} li`))` this will still only be executed one time. 

`allReports: () => element.all(by.css(`${reportListID} li`))` will make it execute every time.